portfolio/src/main/java/com/google/sps/data/CalendarEvent.java

```diff
@@ -0,0 +1,39 @@
+package com.google.sps.data;
+
+import java.time.Instant;
+
+/** Models a calendar event. It can be used for both creation and import flow. */
+public class CalendarEvent {
+  private String name;
+  private final String startTimeString;
+  private final String endTimeString;
+
+  // TODO(hollyyuqizheng): Add an ID field if necessary.
+
+  public CalendarEvent(String name, String startTimeString, String endTimeString) {
+    if (startTimeString == null) {
+      throw new IllegalArgumentException("Event needs a start time");
+    }
+    if (endTimeString == null) {
+      throw new IllegalArgumentException("Event needs an end time");
+    }
+    this.name = name;
+    this.startTimeString = startTimeString;
+    this.endTimeString = endTimeString;
+  }
+
+  public String getName() {
```

Comment:
Is this valid? I didn't see a comment above that indicates which fields are optional.

Reply:
Updated. I am making all three fields mandatory now, and I added the error checking for null name in the constructor.